test/jdk/java/awt/dnd/DragSourceMotionListenerTest.java

```diff
@@ -0,0 +1,250 @@
+            });
+            robot.waitForIdle();
+
+            if (!dsmObj.pointInComponent(robot, srcPoint, source)) {
+                throw new RuntimeException("WARNING: Couldn't locate source panel.");
+            }
+
+            EventQueue.invokeAndWait(() -> {
+                dstOutsidePoint = frame.getLocationOnScreen();
+                d[0] = frame.getSize();
+                dstOutsidePoint.translate(3 * d[0].width / 2, d[0].height / 2);
+                testPoint1.setLocation(dstOutsidePoint);
+             });
+
+
+            EventQueue.invokeAndWait(() -> {
+                dstInsidePoint = target.getLocationOnScreen();
+                d[0] = target.getSize();
+                dstInsidePoint.translate(d[0].width / 2, d[0].height / 2);
+                testPoint2.setLocation(dstInsidePoint);
+            });
+
+            if (!dsmObj.pointInComponent(robot, dstInsidePoint, target)) {
+                throw new RuntimeException("WARNING: Couldn't locate target panel.");
+            }
+
+            robot.mouseMove(srcPoint.x, srcPoint.y);
+            robot.keyPress(KeyEvent.VK_CONTROL);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            for (; !srcPoint.equals(dstOutsidePoint);
+                 srcPoint.translate(sign(dstOutsidePoint.x - srcPoint.x),
+                                    sign(dstOutsidePoint.y - srcPoint.y))) {
+                robot.mouseMove(srcPoint.x, srcPoint.y);
+            }
+
+            for (int i = 0; i < 10; i++) {
+                robot.mouseMove(srcPoint.x, srcPoint.y++);
+            }
+
+            for (;!srcPoint.equals(dstInsidePoint);
+                 srcPoint.translate(sign(dstInsidePoint.x - srcPoint.x),
+                                    sign(dstInsidePoint.y - srcPoint.y))) {
+                robot.mouseMove(srcPoint.x, srcPoint.y);
+            }
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.keyRelease(KeyEvent.VK_CONTROL);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+        if (!passedTest1) {
+            throw new RuntimeException("Failed first test.");
+        }
+
+        if (!passedTest2) {
+            throw new RuntimeException("Failed second test.");
+        }
+    } finally {
+        EventQueue.invokeAndWait(() -> {
+            if (frame != null) {
+                frame.dispose();
+            }
+        });
+    }
+}
+
+    public static int sign(int n) {
+        return Integer.compare(n, 0);
+    }
+
+    public void reset() {
+        clickedComponent = null;
+    }
+
+    public void eventDispatched(AWTEvent e) {
+        if (e.getID() == MouseEvent.MOUSE_RELEASED) {
+            clickedComponent = (Component)e.getSource();
+            mouseReleaseEvent.countDown();
+        }
+    }
+
+    boolean pointInComponent(Robot robot, Point p, Component comp) throws Exception {
+        robot.waitForIdle();
+        reset();
```

Comment:
The `reset` method is called only once from this place. I suggest assigning `clickedComponent = null` right here.

Does it look clearer? You don't have to look into the body of another method.